app/code/Magebit/PageListWidget/Block/Widget/PageList.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * @copyright Copyright (c) 2024 Magebit (https://magebit.com)
+ * @author    Magebit <info@magebit.com>
+ * @license   GNU General Public License ("GPL") v3.0
+ */
+
+declare(strict_types=1);
+
+namespace Magebit\PageListWidget\Block\Widget;
+
+use Magento\Framework\View\Element\Template;
+use Magento\Widget\Block\BlockInterface;
+use Magento\Cms\Model\ResourceModel\Page\CollectionFactory;
+use Magento\Cms\Model\ResourceModel\Page\Collection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * PageList widget block for displaying a list of CMS pages
+ *
+ * @package Magebit\PageListWidget\Block\Widget
+ */
+class PageList extends Template implements BlockInterface
+{
+    /**
+     * @var string
+     */
+    protected $_template = "widget/page_list.phtml";
+
+    /**
+     * PageList constructor.
+     *
+     * @param Template\Context $context
+     * @param CollectionFactory $pageCollectionFactory
+     * @param LoggerInterface $logger
+     * @param array $data
+     */
+    public function __construct(
+        private readonly Template\Context $context,
+        private CollectionFactory $pageCollectionFactory,
+        private LoggerInterface $logger,
+        array $data = []
+    ) {
+        $this->pageCollectionFactory = $pageCollectionFactory;
+        $this->logger = $logger;
+        parent::__construct($context, $data);
```

Comment:
$this->context

Reply:
Not resolved yet. If you use private readonly, it automatically initializes the Object, so it needs to be refered to as $this->context.